File: ympa_fsharp_functions_server/src/model/GetPromosRequest.fs
namespace ympa_fsharp_functions_server.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open ympa_fsharp_functions_server.Model.MechanicsType
open ympa_fsharp_functions_server.Model.PromoParticipationType

module GetPromosRequest =

  //#region GetPromosRequest

  [<CLIMutable>]
  type GetPromosRequest = {
    [<JsonProperty(PropertyName = "participation")>]
    Participation : PromoParticipationType;
    [<JsonProperty(PropertyName = "mechanics")>]
    Mechanics : MechanicsType;
  }

  //#endregion

File: src/main/scala/org/openapitools/server/model/GetPromosRequest.scala
package org.openapitools.server.model


/**
 * Фильтры для получения списка акций.
 *
 * @param participation  for example: ''null''
 * @param mechanics  for example: ''null''
*/
final case class GetPromosRequest (
  participation: Option[PromoParticipationType] = None,
  mechanics: Option[MechanicsType] = None
)

